src/components/Translate.tsx

```diff
@@ -1,14 +1,24 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 
 interface ITranslate {
   baseLanguageValue: string;
   hiddenTranslationValue: string;
+  reset: boolean;
 }
 
-export const Translate: React.FunctionComponent<ITranslate> = ({ baseLanguageValue, hiddenTranslationValue }) => {
+export const Translate: React.FunctionComponent<ITranslate> = ({
+  baseLanguageValue,
```

Comment:
Instead of using `baseLanguageValue` and `hiddenTranslationValue` separately, I would add an object that is matched from translation api. This was mentioned in one of the previous comments as well.